doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+an intra-AS path matching the provided FABRID policy.
+The corresponding intra-AS paths are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path constraints.
+Then it can request the necessary DRKeys from its local control service.
+With this the endhost is able to create FABRID packets and then send them to the border router for furhter forwarding.
+The receiving endhost can then recompute the path validator to verify that the packet was forwarded over this path.
+
+Control plane
+---------------
+
+Control service
+^^^^^^^^^^^^^^^^^
+
+The control plane for FABRID is responsible for parsing FABRID policies into corresponding data structures, and making
+these datastructures available to routers in the same AS, clients in the AS, as well as clients from a remote AS.
+Information for FABRID is also included in the PCBs emitted by each FABRID enabled AS.
+
+The control service introduces a FABRID service with the following endpoints:
+
+- GetMPLSMapIfNecessary (intra-AS)
+- GetRemotePolicyDescription (intra-AS)
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The FABRID service uses the following important data structures:
+
+- SupportedIndicesMap
+    Maps a connection pair consisting of two ConnectionPoints (Type: string, IP: string, Prefix: uint32, InterfaceId: uint16) to a list of policy indices. This map shows for each connection pair which policy indices are supported, which can be one or multiple policies.
+- IndexIdentifierMap
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size. The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers). The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces, a mapping is required from policy index to local and global identifiers. This mapping is provided by this map.
+- IdentifierDescriptionMap
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS. This map maps a local policy identifier to its corresponding description.
+- MPLSMap
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to enforce the policy in the internal network. The MPLSMap data structure stores a mapping between policy indices and the integer MPLS label that is applied to the packet at the border router. Routers periodically fetch this map from the control service. A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
```

Comment:
I don't know how far we should go in detail here, but this is now split up into essentially two maps:
The map which stores the MPLS labels for interface-interface connections, and the map which stores the MPLS label for intf-ip range connections.

Reply:
I think this level of information is not neccessary.